scipy/stats/_axis_nan_policy.py

```diff
@@ -356,7 +356,18 @@ def _axis_nan_policy_factory(tuple_to_result, default_axis=0,
         example, `gmean` accepts as its first argument a sample `a` but
         also `weights` as a fourth, optional keyword argument. In this case, we
         use `n_samples=1` and kwd_samples=['weights'].
+    override : dict, default: {'vectorization': False, 'nan_propagation': True}
+        Pass a dictionary with ``'vectorization': True`` to ensure that the
+        decorator overrides the function's behavior for multimensional input.
+        Use ``'nan_propagation': False`` to ensure that the decorator does not
+        override the function's behavior for ``nan_policy='propagate'``.
+        (See `scipy.stats.mode`, for example.)
     """
+    # Specify which existing behaviors the decorator must override
+    temp = override or {}
+    override = {'vectorization': False,
+                'nan_propagation': True}
```

Comment:
These have been the default behaviors for the decorator: 
1. Don't override vectorization if `axis` is already an argument of the function.
2. Do override behavior for `nan_policy='propagate'` by returning `np.nan` for all elements of outputs.

This isn't appropriate for `mode`. 
1. Even though `mode` had an `axis` argument before, it was done with an N-d loop just like the decorator, so we might as well simplify the implementation and let the decorator take care of vectorization.
2. `np.nan` is not always the correct output when `nan_policy='propagate'`. For example, the mode of `[np.nan, 1, 1]` is `1`.